### ISSUE TITLE
problem with stat time field

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#10189]
llvm-bug.chpl:

``` chapel
extern "struct stat" record chpl_stat {
  var st_mtime: off_t;
}

proc getLastModified(filename: string) : int {
  extern proc stat(filename: c_string, ref chpl_stat): c_int;

  var file_buf: chpl_stat;
  var file_path = filename.c_str();

  if (stat(file_path, file_buf) == 0) {
    return file_buf.st_mtime;
    }
  return -1;
}

writeln(getLastModified("llvm-bug.chpl"));
```

Compile it with `chpl --llvm llvm-bug.chpl`. That gives an internal error.

**Comment by mppf** [comment]
The problem is that in the header configuration we are using, st_mtime is not a field of `struct stat` but is rather a macro:

https://github.com/bminor/glibc/blob/09533208febe923479261a27b7691abef297d604/sysdeps/unix/sysv/linux/x86/bits/stat.h#L84-L104

**Comment by mppf** [comment]
@Spartee @ben-albrecht - fyi

**Comment by mppf** [comment]
I'm not sure if it's practical to get the LLVM back-end to recognize the stat structure. I might rather that the Mason code use sys_stat and we update that to return a structure that we define.

https://github.com/chapel-lang/chapel/blob/e333873e2dd0cbb76554bd61df0b6beadfd3b90d/runtime/src/qio/sys.c#L550-L563

**Comment by gbtitus** [comment]
This is okay with me.  I'm going to mention @dmk42, though, just in case he has feedback regarding the standards.

**Comment by Spartee** [comment]
This bug is being addressed in #10222.

**Comment by dmk42** [comment]
I agree with the proposed direction for a solution.  The evolving POSIX standard has maintained backward compatibility from a C source code point of view, but the way they do it makes things difficult for our LLVM back end to digest.
